### PR TITLE
chore(fluentbit): Update and clean up Fluent Bit values and schema

### PR DESCRIPTION
**Pull request by SanyaKochhar**
**Description**: Updates Fluent Bit related values in charts and schema for cleanliness and clarity

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [x]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

No, no.

### CODE REVIEW

**Review by shashankram — APPROVED** [review]

**Review by nshankar13 — APPROVED** [review]

File: charts/osm/README.md [hunk 1]
```diff
@@ -23,6 +23,12 @@ A Helm chart to install the OSM control plane on Kubernetes
 | OpenServiceMesh.deployPrometheus | bool | `false` |  |
 | OpenServiceMesh.enablePrometheusScraping | bool | `true` | Cannot be false if `deployPrometheus` is true |
 | OpenServiceMesh.envoyLogLevel | string | `"error"` |  |
+| OpenServiceMesh.fluentBit.name | string | `"fluentbit-logger"` |  |
+| OpenServiceMesh.fluentBit.registry | string | `"fluent"` |  |
+| OpenServiceMesh.fluentBit.tag | string | `"1.6.4"` |  |
+| OpenServiceMesh.fluentBit.pullPolicy | string | `"IfNotPresent"` |  |
+| OpenServiceMesh.fluentBit.defaultOutput | string | `"stdout"` |  |
+| OpenServiceMesh.fluentBit.allowCustomOutput | bool | `"true"` |  |
```
Comment (ritazh) [inline charts/osm/README.md:31]: Should this default to `false` since this is only applicable to Log Analytics?

Comment (ritazh) [reply charts/osm/README.md:31]: Given that `Customer_ID` and `Shared_Key` only make sense if the output plugin is log analytics, I think it's more intuitive to only add these configs when `azure` is specified.  For default values, `fluentBit.outputName` should default to `stdout`. For charts that depend on this chart, they can override this `outputName` and provide `workspaceID` and `primaryKey` accordingly. Also `workspaceID` and `primaryKey` should be in values.schema.json as well.

WDYT?

e.g.
```yaml
    [OUTPUT]
      Name    {{ .Values.OpenServiceMesh.fluentBit.outputName }}
      Match   *
      {{- if eq .Values.OpenServiceMesh.fluentBit.outputName "azure" }}
      Customer_ID {{ .Values.OpenServiceMesh.fluentBit.workspaceID }}
      Shared_Key {{ .Values.OpenServiceMesh.fluentBit.primaryKey }}
      {{- end }}
```

Comment (SanyaKochhar) [reply charts/osm/README.md:31]: Oh yes this sounds a lot more intuitive! I think that should work well

Comment (SanyaKochhar) [reply charts/osm/README.md:31]: Addressed in follow up PR: https://github.com/openservicemesh/osm/pull/2116

**Pull request merged**